### ISSUE TITLE
Regression: Button listeners trigger several times on click after reentering screen

### ISSUE THREAD

**Issue by cherrybeans** [Gullskatten/beardblaster#42]
I spent a lot of time debugging this as I thought it was my code when developing, but the problem is in main

When entering Gameplay screen for the second time, and clicking attack, it prints that wizard attacks 3 times, instead of the correct 1 time. If reentering more times, this number multiplies.

I fixed this already in #30, and it worked until 270bf45a3e95f6a90ac5eff355c19df066e9b2d6 (#33) and the issue was back in #34 (d4ccaea2869fc5c9444c621dcae2accb4647dcb4). (I can't build the app from any commits in between these, so maybe we should start squashing before merging into master to avoid this problem. But in fact, I'm struggling to checkout to many of them due to Highscore screen redeclaration error, so not sure if this is just my problem ==> This problem was solved by Build -> Clean project)

**Comment by cherrybeans** [comment]
Any ideas @einaru?

**Comment by cherrybeans** [comment]
After 4 hours of debugging, I eventually tracked down the problem.

`) : BaseScreen(game, batch, assets, camera) {
    private val closeBtn = scene2d.textButton(Nls.close())`

After introducing the fancy styles, we started initializing buttons etc like this. This cannot happen and we get this problem. Buttons and everything else needs to be initialized in the show method (and thus, initScreens() in our setup). Since all of the logs were removed, this problem was hidden. Sometimes when trying to splitting these initializations in classes, the editor suggests to merge them like this, but in this case the editor does not suggest to join, and this is a clue that the two ways are not equal in functionality.
